lib/fs/basicfs_watch.go

```diff
@@ -32,6 +32,12 @@ func (f *BasicFilesystem) Watch(name string, ignore Matcher, ctx context.Context
 		return nil, err
 	}
 
+	// Remove `\\?\` prefix if the path is just a drive letter as a dirty
+	// fix for https://github.com/syncthing/syncthing/issues/5578
+	if runtime.GOOS == "windows" && len(absName) <= 7 && absName[:4] == `\\?\` {
```

Comment:
Panic if absName would happen to be `C:\`, but I guess that can't happen here? Regardless, I'd like to at least check for len > 4...